From: Simon McVittie <smcv@debian.org>
Date: Mon, 9 Oct 2017 09:46:31 +0100
Subject: tests: Use DEB_HOST_ARCH_BITS to skip some tests on 64-bit platforms

Ideally this would use some upstreamable mechanism instead of
modifying the test infrastructure to make dpkg-architecture variables
available, but I don't know the build/test infrastructure well enough
to implement that.

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=905825
Forwarded: no
---
 js/src/tests/non262/Array/regress-157652.js   | 2 +-
 js/src/tests/non262/Array/regress-330812.js   | 2 +-
 js/src/tests/non262/regress/regress-422348.js | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

Index: mozjs-60.8.0/js/src/tests/non262/Array/regress-157652.js
===================================================================
--- mozjs-60.8.0.orig/js/src/tests/non262/Array/regress-157652.js
+++ mozjs-60.8.0/js/src/tests/non262/Array/regress-157652.js
@@ -1,4 +1,4 @@
-// |reftest| skip-if(xulRuntime.XPCOMABI.match(/x86_64|aarch64|ppc64|ppc64le|s390x/)||Android) -- No test results
+// |reftest| skip-if(xulRuntime.debian.DEB_HOST_ARCH_BITS==64||Android) -- No test results
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
Index: mozjs-60.8.0/js/src/tests/non262/Array/regress-330812.js
===================================================================
--- mozjs-60.8.0.orig/js/src/tests/non262/Array/regress-330812.js
+++ mozjs-60.8.0/js/src/tests/non262/Array/regress-330812.js
@@ -1,4 +1,4 @@
-// |reftest| skip-if(xulRuntime.XPCOMABI.match(/x86_64|aarch64|ppc64|ppc64le|s390x/)||Android) -- No test results
+// |reftest| skip-if(xulRuntime.debian.DEB_HOST_ARCH_BITS==64||Android) -- No test results
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
Index: mozjs-60.8.0/js/src/tests/non262/regress/regress-422348.js
===================================================================
--- mozjs-60.8.0.orig/js/src/tests/non262/regress/regress-422348.js
+++ mozjs-60.8.0/js/src/tests/non262/regress/regress-422348.js
@@ -1,4 +1,4 @@
-// |reftest| skip-if(xulRuntime.XPCOMABI.match(/x86_64|aarch64|ppc64|ppc64le|s390x/)) -- On 64-bit, takes forever rather than throwing
+// |reftest| skip-if(xulRuntime.debian.DEB_HOST_ARCH_BITS==64) -- On 64-bit, takes forever rather than throwing
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
